core/formats/sparse_bitmap.hpp

```diff
@@ -145,58 +154,89 @@ class sparse_bitmap_writer {
   size_t bits_[kNumBlocks]{};
   std::vector<block> block_index_;
   uint32_t popcnt_{};
-  uint32_t block_{}; // last flushed block
-}; // sparse_bitmap_writer
+  uint32_t block_{};  // last flushed block
+  doc_id_t prev_value_{};
+  doc_id_t last_in_flushed_block_{};
+  SparseBitmapWriterOptions opts_;
+};
 
-////////////////////////////////////////////////////////////////////////////////
-/// @struct value_index
-/// @brief denotes a position of a value associated with a document
-////////////////////////////////////////////////////////////////////////////////
+// Denotes a position of a value associated with a document.
 struct value_index : document {
-  static constexpr string_ref type_name() noexcept {
-    return "value_index";
+  static constexpr string_ref type_name() noexcept { return "value_index"; }
+};
+
+// Provides an access to previous document before the current one.
+// Undefined after iterator reached EOF.
+class seek_prev : public attribute {
+ public:
+  using seek_prev_f = doc_id_t (*)(const void*);
+
+  static constexpr string_ref type_name() noexcept { return "prev_doc"; }
+
+  constexpr explicit operator bool() const noexcept { return nullptr != func_; }
+
+  constexpr bool operator==(std::nullptr_t) const noexcept {
+    return !static_cast<bool>(*this);
   }
-}; // value_index
 
-////////////////////////////////////////////////////////////////////////////////
-/// @class sparse_bitmap_iterator
-////////////////////////////////////////////////////////////////////////////////
+  constexpr bool operator!=(std::nullptr_t) const noexcept {
+    return !(*this == nullptr);
+  }
```

Comment:
Since C++20 no needed?
![image](https://user-images.githubusercontent.com/32071355/171875954-bb4f4ecb-3f64-49e1-9fc8-d4ef0b493cad.png)

Reply:
I think the implementation differs from the default behavior.

Reply:
Done